File: backend/components/metiers/administration_publique/tests/utils/core/test_utils.js
// test_utils.js - Tests unitaires JS pour les utilitaires Threed
const audit = require('../utils/audit');
const i18n = require('../utils/i18n');

describe('Audit Threed', () => {
  it('doit retourner un score correct', () => {
    const result = audit.auditThreed({ status: 'active' });
    expect(result.score).toBe(97.0);
  });
});

describe('i18n Threed', () => {
  it('doit traduire en français', () => {
    expect(i18n.i18n('Test', 'fr')).toMatch('[FR]');
  });
  it('doit traduire en anglais', () => {
    expect(i18n.i18n('Test', 'en')).toMatch('[EN]');
  });
  it('doit fallback sur la valeur brute', () => {
    expect(i18n.i18n('Test', 'es')).toBe('Test');
  });
});

File: backend/components/metiers/administration_publique/tests/utils/core/test_utils.py
from .core_utils import audit_log
from ..i18n.i18n_utils import translate

def auditThreed(payload):
    # Simule un audit ultra basique pour test
    return {'score': 97.0, 'payload': payload}

def i18n(msg, lang):
    return translate(msg, lang)

def test_audit_threed():
    result = auditThreed({'status': 'active'})
    assert 'score' in result
    assert result['score'] == 97.0

def test_i18n():
    assert '[FR]' in i18n('Test', 'fr')
    assert '[EN]' in i18n('Test', 'en')
    assert '[DE]' in i18n('Test', 'de')
